controllers/articles.js

```diff
@@ -267,4 +281,125 @@ export default class ArticleController {
       });
     }
   }
+
+  /**
+ * @description - Get all articles
+ * @static
+ * @param {Object} req - the request object
+ * @param {Object} res - the response object
+ * @memberof ArticleController
+ * @returns {Object} class instance
+ */
+  static async getAllArticles(req, res) {
+    try {
+      let {
+        query: {
+          offset, limit
+        }
+      } = req;
+      offset = Number(offset) || 0;
+      limit = Number(limit) || 10;
+      const results = await Article.findAndCountAll({
+        where: {},
+        offset,
+        limit,
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['username', 'bio', 'name']
+          },
+          {
+            model: Category,
+            as: 'category',
+            attributes: ['categoryName']
+          },
+          {
+            model: Ratings,
+          }
+        ]
+      });
+      const { count } = results;
+      const meta = Paginate({ count, limit, offset });
+      return res.status(200).json({
+        success: true,
+        results,
+        ...meta
+      });
+    } catch (error) {
+      return res.status(500).json({
+        succes: false,
+        errors: ['Oops, something wrong occured.']
+      });
+    }
+  }
+
+  /**
+  * @description - Get a specific number of articles with criteria
+  * @static
+  * @param {Object} req - the request object
+  * @param {Object} res - the response object
+  * @memberof ArticleController
+  * @returns {Object} class instance
+  */
+  static async getArticlesByHighestField(req, res) {
+    try {
+      const {
+        query: {
+          amount, type
+        }
+      } = req;
+      let articles;
+      if (type === 'latest') {
+        articles = await Article.findAll({
+          where: {},
+          limit: Number(amount) || 5,
+          order: [['createdAt', 'ASC']],
+          include: [
+            {
+              model: User,
+              as: 'author',
+              attributes: ['username', 'bio', 'name']
+            },
+          ]
+        });
+      } else if (type === 'comments' || type === 'ratings') {
+        const results = await Article.findAll({
+          where: {},
+          limit: Number(amount) || 5,
+          include: [
+            {
+              model: User,
+              as: 'author',
+              attributes: ['username', 'bio', 'name']
+            },
+            {
+              model: Comment
+            },
+            {
+              model: Ratings,
+            }
+          ]
+        });
+        if (type === 'comments') {
+          articles = results
+            .sort((a, b) => b.Comments.length - a.Comments.length)
+            .slice(0, amount);
+        } else {
+          articles = results
+            .sort((a, b) => b.Ratings.length - a.Ratings.length)
+            .slice(0, amount);
+        }
+      }
+      return res.status(200).json({
+        success: true,
+        articles
```

Comment:
No response message

Reply:
There's no need for a message if the request was successful @Dubby20

Reply:
If a request is successful you have to send a message to let the user know the status of their request.